contracts/signature/ExecutionTracker.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0 <0.9.0;
+
+/// @title ExecutionTracker - A contract that keeps track of executed and invalidated hashes
+contract ExecutionTracker {
+    mapping(address => mapping(bytes32 => bool)) public executed;
```

Comment:
We could use one namespace for all modules if we include the module address in the data we use to create the hash.
```js
keccak256(abi.encode(MODULE_TX_TYPEHASH, keccak256(data), module, salt))
```
Although I guess this would require also appending the module address along with the signature and the nonce, no?

Reply:
I think i'd prefer to use the same hash that gets signed. 

This is for clarity, but also such that calls to invalidate, or calls to check whether previously executed, can done by staticcalling moduleTxHash, and then using that hash output

Reply:
okay I get it now. This would require appending the module to the tx.

Reply:
Not sure which option would be more cost effective.

Reply:
Let's stick with the current structure then.